pkg/distributor/distributor.go

```diff
@@ -597,32 +575,23 @@ func (d *Distributor) sendSamples(ctx context.Context, ingester ring.IngesterDes
 
 	req := client.WriteRequest{
 		Timeseries: timeseries,
+		Metadata:   metadata,
 	}
 	_, err = c.Push(ctx, &req)
 
-	ingesterAppends.WithLabelValues(ingester.Addr, typeSamples).Inc()
-	if err != nil {
-		ingesterAppendFailures.WithLabelValues(ingester.Addr, typeSamples).Inc()
-	}
-	return err
-}
-
-func (d *Distributor) sendMetadata(ctx context.Context, ingester ring.IngesterDesc, metadata []*client.MetricMetadata) error {
-	h, err := d.ingesterPool.GetClientFor(ingester.Addr)
-	if err != nil {
-		return err
-	}
-	c := h.(ingester_client.IngesterClient)
-
-	req := client.WriteRequest{
-		Metadata: metadata,
+	if len(metadata) > 0 {
+		ingesterAppends.WithLabelValues(ingester.Addr, typeSamples).Inc()
```

Comment:
In the metrics tracking you've inverted `typeSamples` with `typeMetadata`. Please enrich unit tests to assert on tracked metrics. This should be spot by unit tests.

Reply:
Good catch, have added the tests.